### PR TITLE
Jliu/merge schema

### PR DESCRIPTION
**Pull request by johnliucn925**
make sure that query, mutation and subscription from each services will be merged into overall schema

### CODE REVIEW

File: src/main/java/graphql/nadel/DefinitionRegistry.java [hunk 1]
```diff
@@ -42,17 +44,30 @@ public SchemaDefinition getSchemaDefinition() {
         return (SchemaDefinition) definitionsByClass.get(SchemaDefinition.class).get(0);
     }
 
+    public Map<OperationType, ObjectTypeDefinition> getOperationTypes(){
+        Map<OperationType, ObjectTypeDefinition > opsTypesMap = new HashMap<>();
+
+        Arrays.stream(OperationType.values()).forEach(opsType ->
+                opsTypesMap.put(opsType, getOpsType(opsType.getOpsType(), opsType.getDisplayName())));
+
+       return opsTypesMap;
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/DefinitionRegistry.java]: Personally I think for simple`for` loops are better than streams.  I guess its a bit personal choice but I think it reads nicer

```
for (opsType : OperationType.values()) {
  opsTypesMap.put(opsType, getOpsType(opsType.getOpsType(), opsType.getDisplayName());
}
```

File: src/main/java/graphql/nadel/DefinitionRegistry.java [hunk 2]
```diff
@@ -42,17 +44,30 @@ public SchemaDefinition getSchemaDefinition() {
...
+    }
+
     public ObjectTypeDefinition getQueryType() {
+        return getOpsType(OperationType.QUERY.getOpsType(), OperationType.QUERY.getDisplayName());
+    }
+
+    private ObjectTypeDefinition getOpsType(String typeName, String typeDisplay ) {
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/DefinitionRegistry.java]: naming nitpick but typeName sounds like a "graphql type".   "operationTypeName / operationDisplayName" might be better names

File: src/main/java/graphql/nadel/OperationType.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package graphql.nadel;
+
+public enum OperationType {
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/OperationType.java]: Again the term "type" in graphql tends to mean the graphql types.

I would just call this "Operation" and have "name" and "displayName" as the attributes

File: src/main/java/graphql/nadel/DefinitionRegistry.java (hunk 1 above)
Comment (tsroka) [reply src/main/java/graphql/nadel/DefinitionRegistry.java]: Or use stream and Collectors.toMap

File: src/main/java/graphql/nadel/OperationType.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    QUERY("query", "Query"),
+    MUTATION("mutation", "Mutation"),
+    SUBSCRIPTION("subscription", "Subscription");
+
+    private String opsType;
```
Comment (tsroka) [inline src/main/java/graphql/nadel/OperationType.java]: maybe it should be `name`? type is already a part of the enum name

**Pull request merged**